### PR TITLE
UHM-4932 & UHM-4967

### PR DESCRIPTION
**Pull request by mogoodrich**
Handles both these tickets:

https://pihemr.atlassian.net/browse/UHM-4932

https://pihemr.atlassian.net/browse/UHM-4967

I've run the whole migration soup-to-nuts with most of these changes.  I did make a few last-minute tweaks, will run again this afternoon/evening before merging.

### CODE REVIEW

File: etl/src/main/java/org/pih/hivmigration/etl/sql/EncounterMigrator.groovy [hunk 1]
```diff
@@ -177,14 +181,12 @@ class EncounterMigrator extends SqlMigrator {
...
+            INSERT INTO hivmigration_data_warnings (openmrs_patient_id, openmrs_encounter_id, field_name, warning_type, warning_details)
             SELECT p.person_id,
                    e.encounter_id,
                    'encounter_date',
-                   CONCAT('Encounter date is null. `date_created` used as encounter date. ',
-                          'Source encounter_id ', e.source_encounter_id,
-                          '. Source patient_id ', e.source_patient_id,
-                          '. Source encounter_type ', e.source_encounter_type, '.')
+                   'Encounter date is null. `date_created` used as encounter date.',
+                   CONCAT('Source encounter_type :', e.source_encounter_type)
```
Comment (mogoodrich) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/EncounterMigrator.groovy:189]: Removed source_encounter_id and source_patient_id as they now have their own columns.

**Review by cioan — APPROVED** [review]
Thanks @mogoodrich ! Looks good!

**Review by brandones — APPROVED** [review]
Very nice!

**Comment by brandones** [comment]
@mogoodrich Can we put this in? I'm adding data warnings and it would be great to be able to just add them in the new format rather than resolving later.

**Comment by mogoodrich** [comment]
Will merge now, thanks @brandones ... will you go back and fix the other data warnings you just merged in then? :)

**Comment by brandones** [comment]
Thanks Mark! Yep: https://github.com/PIH/hivmigration/pull/67

**Pull request merged**